src/src/com/microsoft/aad/adal/TokenCacheItem.java

```diff
@@ -66,39 +66,76 @@
     private String mFamilyClientId;
 
     /**
-     * Construct default cache item.
+     * Default constructor for cache item.
      */
-    public TokenCacheItem() {
-
+    public TokenCacheItem() {}
+    
+    TokenCacheItem(final TokenCacheItem tokenCacheItem) {
```

Comment:
when removing token when getting invalid_grant, and if i use MRRT to do the refresh token request, i also need to clean up regular rt token item. So i use the copy constructor to create a new item, and set resource into the item.